pkg/types/openstack/machinepool.go

```diff
@@ -77,3 +95,9 @@ type RootVolume struct {
 	// +optional
 	Zones []string `json:"zones,omitempty"`
 }
+
+// ServerGroupPolicy is the policy to be applied to an OpenStack Server Group.
+//
+// +kubebuilder:validation:Enum=affinity;soft-affinity;anti-affinity;soft-anti-affinity
```

Comment:
Need to accept an empty string, too.
```suggestion
// +kubebuilder:validation:Enum="";affinity;soft-affinity;anti-affinity;soft-anti-affinity
```